### ISSUE TITLE
Base should be baked into countertop

### ISSUE THREAD

**Issue by slifty** [tvkitchen/countertop#154]
Especially once #153 is complete I think it will make a lot of sense for all of the "base" and "core" functionality to get merged into `countertop`

This means the flow for a developer would be to import countertop and do whatever they want to do without needing to worry about importing a bunch of random "base-error" or "base-interface" junk.  Countertop will truly be the entry point not just for appliance users but for appliance developers.  This will also make it easier to make sure a countertop + appliance pair is in sync in terms of what version of constants exist instead of updating a bunch of redundant dependencies.

**Comment by slifty** [comment]
I don't think there's any good reason to import the `errors` package / classes.  Most of those were just there to support... type checks.
